lib/wallet/src/Cardano/Wallet/Primitive/AddressDiscovery/Sequential.hs

```diff
@@ -488,33 +491,35 @@ decoratePath st r ix = NE.fromList
 -- addresses on the internal chain anywhere in the available range.
 instance SupportsDiscovery n k => IsOurs (SeqState n k) Address where
     isOurs addrRaw st =
-        -- FIXME LATER: Check that the network discrimant of the type
-        -- is compatible with the discriminant of the Address!
-        case paymentKeyFingerprint addrRaw of
-            Left _ -> (Nothing, st)
-            Right addr ->
-                let (internalIndex, !internalPool') =
-                        lookupAddress addr (internalPool st)
-                    (externalIndex, !externalPool') =
-                        lookupAddress addr (externalPool st)
-
-                    ours = (decoratePath st UtxoExternal <$> externalIndex)
-                       <|> (decoratePath st UtxoInternal <$> internalIndex)
-
-                    pendingChangeIxs' =
-                        case internalIndex of
-                            Nothing -> pendingChangeIxs st
-                            Just ix ->
-                                dropLowerPendingIxs ix (pendingChangeIxs st)
-                in
-                    ( ours `deepseq` ours
-                    , st { internalPool = internalPool'
-                         , externalPool = externalPool'
-                         , pendingChangeIxs =
-                               pendingChangeIxs' `deepseq` pendingChangeIxs'
-                         }
-                    )
+        if networkTag == networkVal @n then
```

Comment:
I believe this is what messes up https://github.com/input-output-hk/cardano-wallet/pull/3791#issuecomment-1477862242 in the case of Byron addresses